cmd/tailscale/cli/web.go

```diff
@@ -224,8 +224,8 @@ func qnapAuthnQtoken(r *http.Request, user, token string) (string, *qnapAuthResp
 		"user":   []string{user},
 	}
 	u := url.URL{
-		Scheme:   r.URL.Scheme,
-		Host:     r.URL.Host,
+		Scheme:   "http",
```

Comment:
Should this be conditional on the qnap version?

Reply:
In https://github.com/tailscale/tailscale-qpkg/tree/dgentry I gave up on QTS 4 support after QNAP pushed 5.0 to the unit even though I kept refusing it. I don't have a way to test it, there are a number of things which don't work on QTS 4.